### ISSUE TITLE
Create state diagram for Beam FSM flows

### ISSUE THREAD

**Issue by rajnikantsh** [LBNL-UCB-STI/beam#1868]

**Comment by rajnikantsh** [comment]
![ChoosesMode](https://user-images.githubusercontent.com/5759657/58720419-bf373700-83ef-11e9-9686-3b0387b69af1.png)
![DrivesVehicleFSM](https://user-images.githubusercontent.com/5759657/58720420-bf373700-83ef-11e9-8b3e-a6e4cb149004.png)
![DriveVehiclesFSM](https://user-images.githubusercontent.com/5759657/58720421-bf373700-83ef-11e9-81a3-0812c5500916.png)
![RideHailAgentFSM](https://user-images.githubusercontent.com/5759657/58720423-bfcfcd80-83ef-11e9-981c-7eca0498e423.png)
![TransistDriverFSM](https://user-images.githubusercontent.com/5759657/58720424-bfcfcd80-83ef-11e9-803a-690de6203892.png)

**Comment by michaz** [comment]
Nice! Note that the `StateTimeout`s are a hack -- They are not _actually_ a timeout, but _I could not find a better way to go from state A to state C _through_ state B (reporting the transition) without an additional event.

If we make things explicit, with diagrams, we should try and remove this, otherwise it confuses people. It was okay while it was hidden away. ;-)

**Comment by rajnikantsh** [comment]
Hi @michaz . can you please check these diagram. We have removed state timeout . are they more clear to previous ones ?

![ChoosesMode](https://user-images.githubusercontent.com/5759657/59056806-a5499880-88b6-11e9-8324-f92c2ac13f60.png)
![ChoosesParking](https://user-images.githubusercontent.com/5759657/59056807-a5499880-88b6-11e9-8dc4-9a2fce1194c6.png)
![PersonAgent](https://user-images.githubusercontent.com/5759657/59056808-a5e22f00-88b6-11e9-9067-fdf604413f0f.png)

**Comment by kirillm-nor** [comment]
Hi @rajnikantsh, could you also combine states to high-level cluster to show inter mode transition, like
_______________PersonAgent_________          _____DrivingVehicle____
| ProcessingNextLegOrStartActivity | --> |     WaitingToDrive       |
|_____________________________________ |       | _______________________|      
It will give better picture of the whole agent stages transaction.

**Comment by michaz** [comment]
@rajnikantsh Yes, thanks! Consider removing the event tag for the `IntermediateTransitionState` entirely, making those arrows empty. If you don't like that, please call it `IntermediateTransition`: The _boxes_ are states, not the arrows.

**Comment by rajnikantsh** [comment]
Updated and Unified states diagram
![ChoosesMode](https://user-images.githubusercontent.com/5759657/59223904-b7934180-8bea-11e9-9287-843265a88bd8.png)
![ChoosesParking](https://user-images.githubusercontent.com/5759657/59223905-b7934180-8bea-11e9-8705-eb4ac63f76c5.png)
![PersonAgent](https://user-images.githubusercontent.com/5759657/59223906-b82bd800-8bea-11e9-98f3-3611db60a32b.png)
![UnifiedStateFSM](https://user-images.githubusercontent.com/5759657/59223908-b82bd800-8bea-11e9-84a5-f104475602b2.png)

**Comment by Xuan-1998** [comment]
@rajnikantsh Did we have any synchronization for state transition?

**Comment by rajnikantsh** [comment]
@Xuan-1998 check this https://plantuml.com/
